includes/payment-methods/class-upe-payment-method.php

```diff
@@ -192,10 +192,48 @@ public function has_domestic_transactions_restrictions() {
 	 * @return bool
 	 */
 	public function is_enabled_at_checkout( string $account_country, bool $skip_limits_per_currency_check = false ) {
-		if ( $this->is_subscription_item_in_cart() || $this->is_changing_payment_method_for_subscription() ) {
+		// Check if we're in a subscription context (cart checkout, changing payment method, or renewal).
+		$is_subscription_context = $this->is_subscription_item_in_cart() || $this->is_changing_payment_method_for_subscription();
+
+		// Also check if we're on the order-pay page for a renewal order.
+		if ( ! $is_subscription_context && is_wc_endpoint_url( 'order-pay' ) && function_exists( 'wcs_order_contains_renewal' ) ) {
+			$order = wc_get_order( absint( get_query_var( 'order-pay' ) ) );
+			if ( $order && wcs_order_contains_renewal( $order ) ) {
+				$is_subscription_context = true;
+			}
+		}
+
+		if ( $is_subscription_context ) {
+			// Check if we're changing payment method for a specific subscription.
+			if ( $this->is_changing_payment_method_for_subscription() && function_exists( 'wcs_get_subscription' ) ) {
+				$subscription_id = absint( $_GET['change_payment_method'] ?? 0 ); // phpcs:ignore WordPress.Security.NonceVerification
+				if ( $subscription_id ) {
+					$subscription = wcs_get_subscription( $subscription_id );
+					if ( $subscription ) {
+						// Allow non-reusable methods ONLY if this specific subscription is manual.
+						if ( $subscription->is_manual() ) {
+							return true;
+						}
+						// For automatic subscriptions, only allow reusable methods.
+						return $this->is_reusable();
+					}
+				}
+			}
+
+			// Allow non-reusable payment methods if manual renewals are accepted globally.
+			if ( $this->are_manual_renewals_accepted() ) {
```

Comment:
Related to my comment below, should we also check if the amount is within the payment method limits?

Reply:
I fixed the conditions check in ff030c79b.